dspace-oai/src/main/java/org/dspace/xoai/app/XOAIItemCompilePlugin.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.xoai.app;
+
+import com.lyncode.xoai.dataprovider.xml.xoai.Metadata;
+import org.dspace.content.Item;
+import org.dspace.core.Context;
+
+
+/**
+ * Used to enrich item.compile field description
+ *
+ */
+public interface XOAIItemCompilePlugin {
+
+    public Metadata additionalMetadata(Context context, Metadata metadata, Item item);
```

Comment:
javadoc is needed here.
what about:
This method allows XOAIItemCompilePlugins to add content to the xoai document generated for the item.
metadata: the basic xoai representation of the item that can be manipulated by the plugin
item: the dspace item to index